### DISCUSSION TITLE
Feature Request: not following redirects

### DISCUSSION THREAD

**Discussion by EtienneBruines** [gavv/httpexpect#136] [Q&A] [answered]
Hi!

Sometimes, I'd want to be able to "not follow" redirects, such as `http.StatusFound`. A lot of my pages are behind a login, and checking whether or not the page works, by saying `.expect(http.StatusOK)` would be misleading; since the server returned a `http.StatusFound` instead, and now redirected the client to the login page. 

I'm currently using this workaround, but it's doesn't really feel right:

``` go

        url := e.GET("/secure/").
            Expect().
            Status(http.StatusOK).
            Raw().Request.URL.RequestURI()
        assert.Equal(t, "/secure", url, "We are authenticated and trying to access /secure")
```

I'd love to be able to do something like:

``` go
e.GET("/secure").
    NoFollow().
    Expect().
    Status(http.StatusOK)
```

**Comment by EtienneBruines** [comment, marked as answer]
Nevermind, this can be done by changing the `http.Client`:

``` go
        Client: &http.Client{
            Transport: httpexpect.NewFastBinder(handler),
            CheckRedirect: func(req *http.Request, via []*http.Request) error {
                return http.ErrUseLastResponse
            },
            Jar: httpexpect.NewJar(),
        },
```

I followed this [Stack Overflow](http://stackoverflow.com/questions/23297520/how-can-i-make-the-go-http-client-not-follow-redirects-automatically/38150816#38150816) question, but apparently, the answers were outdated. 

Apologies for the fuss.

**Comment by gavv** [comment]
Thanks for sharing this. I've also seen that question. Actually, `ErrUseLastResponse` seems to be the best answer :)

**Comment by pas256** [comment]
I know this is old so apologies, but I do not see how to use the code @EtienneBruines posted within a test. How do I modify the underlying `Client`?

**Comment by EtienneBruines** [comment]
@pas256 It's been a while, so I have no idea. 

But, looking at the source code of one of the tests:

https://github.com/gavv/httpexpect/blob/master/e2e_redirect_test.go#L58
```go
func TestE2ERedirectBinderStandard(t *testing.T) {
	handler := createRedirectHandler()

	testRedirectHandler(WithConfig(Config{
		BaseURL:  "http://example.com",
		Reporter: NewAssertReporter(t),
		Client: &http.Client{
			Transport: NewBinder(handler),
		},
	}))
}
```

It seems like you can use `WithConfig` to set it. See also [WithConfig](https://github.com/gavv/httpexpect/blob/3dd83fcd5f57a3fe3c8b2674b2ec2c3475301afa/expect.go#L298)

**Comment by gavv** [comment]
Yes, you can use WithConfig or WithClient:

* https://github.com/gavv/httpexpect#custom-config
* https://github.com/gavv/httpexpect#per-request-client-or-handler
